pulsar-broker/src/main/java/org/apache/pulsar/broker/service/schema/SchemaRegistryServiceImpl.java

```diff
@@ -350,15 +350,9 @@ private void checkCompatible(SchemaAndMetadata existingSchema, SchemaData newSch
                                  SchemaCompatibilityStrategy strategy) throws IncompatibleSchemaException {
         SchemaHash existingHash = SchemaHash.of(existingSchema.schema);
         SchemaHash newHash = SchemaHash.of(newSchema);
-        SchemaData existingSchemaData = existingSchema.schema;
-        if (newSchema.getType() != existingSchemaData.getType()) {
-            throw new IncompatibleSchemaException(String.format("Incompatible schema: "
-                            + "exists schema type %s, new schema type %s",
-                    existingSchemaData.getType(), newSchema.getType()));
-        }
         if (!newHash.equals(existingHash)) {
```

Comment:
I think the compatibility check should be cohesive to checkCompatible instead of scattered outside, so I suggest to remove the outer check

Reply:
Yeah that's a little bit cleaner! But the "type" check inside should be at the beginning. Thanks for taking the time to review it.